src/jdmtool/data_card/detect.py

```diff
@@ -85,15 +100,15 @@ def open_programming_device() -> Generator[ProgrammingDevice, None, None]:
                 else:
                     raise ProgrammingException("Could not find the new device!")
 
-                print("Writing stage 2 firmware...")
+                # check version and write stage 2 if required
                 with _open_usb_device(usbdev) as handle:
                     writer = GarminFirmwareWriter(handle)
                     writer.init_stage2()
                     writer.write_firmware_stage2()
 
                 print("Re-scanning devices...")
                 for _ in range(5):
-                    time.sleep(0.5)
+                    time.sleep(2)
```

Comment:
0.5->2 will slow down the initialization if the actual time ends up being e.g. 2.5s. Change `range(5)` to `range(20)` instead?